main/exec.go

```diff
@@ -25,11 +26,74 @@ func Exec(ctx *log.Context, cmd, workdir string, stdout, stderr io.WriteCloser,
 	defer stdout.Close()
 	defer stderr.Close()
 
+	scriptPath := cmd
+	scriptPathDirectory := filepath.Dir(cmd)
+
 	commandArgs, err := SetEnvironmentVariables(cfg)
-	cmd = cmd + commandArgs // Add command args if any
+	// Add command args if any. Unnamed arguments go in 'commandArgs'. Named arguments are set as environment variables so the'd be available within the script.
+	cmd = cmd + commandArgs
 
 	//executionMessage := ""   // TODO: return
 	exitCode := 0 // TODO: return exit code and execution state
+
+	if cfg.publicSettings.RunAsUser != "" {
```

Comment:
Add a unit test for this section that checks whether the created script file and the directory that contains it is owned by the runAsUser

Reply:
I'll soon add the unit tests [Task 16026436](https://msazure.visualstudio.com/One/_workitems/edit/16026436): [RCv2 Linux] - Add unit tests for RunAs